robottelo/hosts.py

```diff
@@ -226,16 +226,33 @@ def satellite(self):
             self._satellite = Satellite()
         return self._satellite
 
+    @property
+    def _sat_host_record(self):
+        """Provide access to this host's Host record if it exists."""
+        host = None
+        with contextlib.suppress(IndexError):
+            host = self.satellite.api.Host().search(query={'search': self.hostname})[0]
```

Comment:
Can you add `logger.warning/debug` here instead of in `_delete_host_record`, when host record is not found.

Reply:
addressed in ac03928e4